react/features/filmstrip/components/native/TileView.js

```diff
@@ -0,0 +1,324 @@
+// @flow
+
+import React, { Component } from 'react';
+import {
+    ScrollView,
+    TouchableWithoutFeedback,
+    View
+} from 'react-native';
+import { connect } from 'react-redux';
+
+import {
+    getNearestReceiverVideoQualityLevel,
+    setMaxReceiverVideoQuality
+} from '../../../base/conference';
+import {
+    DimensionsDetector,
+    isNarrowAspectRatio,
+    makeAspectRatioAware
+} from '../../../base/responsive-ui';
+
+import Thumbnail from './Thumbnail';
+import styles from './styles';
+
+/**
+ * The type of the React {@link Component} props of {@link TileView}.
+ */
+type Props = {
+
+    /**
+     * The participants in the conference.
+     */
+    _participants: Array<any>,
+
+    /**
+     * Invoked to update the receiver video quality.
+     */
+    dispatch: Dispatch<*>,
+
+    /**
+     * Callback to invoke when tile view is tapped.
+     */
+    onClick: Function
+};
+
+/**
+ * The type of the React {@link Component} state of {@link TileView}.
+ */
+type State = {
+
+    /**
+     * The available width for {@link TileView} to occupy.
+     */
+    height: number,
+
+    /**
+     * The available height for {@link TileView} to occupy.
+     */
+    width: number
+};
+
+const TILE_ASPECT_RATIO = 1;
+
+/**
+ * Implements a React {@link Component} which represents the filmstrip on
+ * mobile/React Native.
+ *
+ * @extends Component
+ */
+class TileView extends Component<Props, State> {
+    state = {
+        height: 0,
+        width: 0
+    };
+
+    /**
+     * Initializes a new {@code TileView} instance.
+     *
+     * @param {Object} props - The read-only properties with which the new
+     * instance is to be initialized.
+     */
+    constructor(props: Props) {
+        super(props);
+
+        // Bind event handler so it is only bound once per instance.
+        this._onDimensionsChanged = this._onDimensionsChanged.bind(this);
+    }
+
+    /**
+     * Updates the receiver video quality.
```

Comment:
It does that for sure, but wouldn't it be more flexible to use a doc related to ```componentDidMount``` so then we don't need to change it if we add more logic? (same below)